controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -540,6 +541,29 @@ private void registerCname(String name, String targetName) {
         }
     }
 
+    /** Returns the endpoints of the clusters in the deployment, or empty if the request fails. */
+    public Optional<Map<ClusterSpec.Id, URI>> clusterEndpoints(DeploymentId id) {
```

Comment:
Should be `Map<ClusterSpec.Id, URI>`. There's never a good reason to have `Optional<SomeCollection>`...

Reply:
I used the pattern we already had for getting endpoints from the routing layer, which I think may have used this to discriminate between exceptional/no reply, and just "no endpoints", at some point ... but I don't really see any need for that in the current code. So I guess I can rather clean up that as well :)